routes/Program/controllers.js

```diff
@@ -9,24 +9,140 @@ const router = express.Router();
 router.post('/', checkSchema(programSchema), async (req, res, next) => {
    const {
       name,
-      collegeId
+      collegeId,
+      degree,
+      language,
+      type,
+      accredited,
+      duration,
+      organization,
+      credits,
+      totalPlaces,
+      taxFreePlaces,
+      taxPlaces,
+      admissionType,
+      courses,
+      competencies
    } = req.body;
 
    try {
-      await service.create(name, ObjectId(collegeId));
+      await service.create(name, ObjectId(collegeId), degree, language, type, accredited, duration, organization, credits,
+          totalPlaces, taxFreePlaces, taxPlaces, admissionType, courses, competencies);
 
       res.status(201).end();
    } catch (err) {
       next(err);
    }
 });
 
-router.get('/', checkSchema(querySchema), async (req, res, next) => {
+router.get('/:id', checkSchema(querySchema), async (req, res, next) => {
+   const {
+      id
+   } = req.params;
+
    try {
-      res.status(200).json(await service.getAll(req.query));
+      res.status(200).json(await service.getById(id, querySchema));
```

Comment:
Same as above, no need for query (which i think should be `req.query`, not `querySchema`) in a get by id.

Reply:
done